toolkit/featureforms/src/androidTest/java/com/arcgismaps/toolkit/featureforms/ComboBoxFieldTests.kt

```diff
@@ -0,0 +1,444 @@
+        comboBoxField.assertTextEquals(formElement.label, formElement.value.value)
+        // find the clear text node within its children
+        val clearButton = comboBoxField.onChildWithContentDescription(clearTextSemanticLabel)
+        // validate the clear icon is visible
+        clearButton.assertIsDisplayed()
+        // clear the value
+        clearButton.performClick()
+        // assert "no value" placeholder is visible
+        // assertTextEquals matches the Text(the label) and Editable Text (the placeholder here
+        // due to the use of the PlaceHolderTransformation)
+        comboBoxField.assertTextEquals(
+            formElement.label,
+            input.noValueLabel.ifEmpty { context.getString(R.string.no_value) }
+        )
+    }
+
+    /**
+     * Test case 3.2:
+     * Given a ComboBoxField with no pre-existing value, description and a no value label
+     * When the field is observed
+     * Then the ComboBoxField shows the noValueLabel and the options menu icon is visible
+     * https://devtopia.esri.com/runtime/common-toolkit/blob/main/designs/Forms/FormsTestDesign.md#test-case-32-no-pre-existing-value-no-value-label-options-button
+     */
+    @Test
+    fun testNoValueAndNoValueLabel() {
+        val formElement = getFormElementWithLabel("Combo Integer")
+        val input = formElement.input as ComboBoxFormInput
+        // find the field with the the label
+        val comboBoxField = composeTestRule.onNodeWithText(formElement.label)
+        comboBoxField.printToLog("TAG")
+        // assert it is displayed and not focused
+        comboBoxField.assertIsDisplayed()
+        comboBoxField.assertIsNotFocused()
+        // find the child node with the description semantic label
+        val descriptionNode = comboBoxField.onChildWithContentDescription(descriptionSemanticLabel)
+        val hasDescriptionMatcher = hasText(formElement.description)
+        // validate the correct description is visible
+        assert(hasDescriptionMatcher.matches(descriptionNode.fetchSemanticsNode())) {
+            "Failed to assert the following: ${hasDescriptionMatcher.description}"
+        }
+        // assert "no value" placeholder is visible
+        // assertTextEquals matches the Text(the label) and Editable Text (the placeholder here
+        // due to the use of the PlaceHolderTransformation)
+        comboBoxField.assertTextEquals(
+            formElement.label,
+            input.noValueLabel.ifEmpty { context.getString(R.string.no_value) }
+        )
+        // validate that the options icon is visible
+        val optionsIconNode =
+            comboBoxField.assertContentDescriptionContains(optionsIconSemanticLabel)
+        optionsIconNode.assertIsDisplayed()
+    }
+
+    /**
+     * Test case 3.3:
+     * Given a ComboBoxField with a pre-existing value, description and a no value label
+     * When the ComboBoxField is tapped
+     * Then the ComboBoxDialog is shown with all coded values AND
```

Comment:
The test design doesn't specify that the dialog should show all coded values, and this test doesn't assert it either. I think this part of the comment can be removed. 

It seems like a good subject for a runtimecore unit test to test that all coded values are in the list, but not something that we need to assert in the toolkit.
```suggestion
     * Then a ComboBoxDialog is shown with AND
```